packages/go_router/lib/src/matching.dart

```diff
@@ -97,8 +98,11 @@ class RouteMatchList {
 
   /// Removes the last match.
   void pop() {
+    if (_matches.last.route is GoRoute) {
+      final GoRoute route = _matches.last.route as GoRoute;
+      _uri = _uri.replace(path: removePatternFromPath(route.path, _uri.path));
```

Comment:
It might be better to keep a single source of truth - the `List<RouteMatch>` should be all that we need to compute the current url - otherwise we risk the URI getting out of sync with the list of matches.

We could make `uri` a getter that produces the URI based on the current match list:
```
Uri get uri => _computeUri(matches);
```
```

Reply:
I agree, I am worrying about the performance though. let me see what i can do.

Reply:
That's fair - as written this code looks good.

Reply:
I decided to move forward at the current implementation, we can revisit if this does become a problem.